sympy/core/add.py

```diff
@@ -530,8 +530,17 @@ def _eval_is_algebraic_expr(self, syms):
         (a.is_finite for a in self.args), quick_exit=True)
     _eval_is_hermitian = lambda self: _fuzzy_group(
         (a.is_hermitian for a in self.args), quick_exit=True)
-    _eval_is_integer = lambda self: _fuzzy_group(
-        (a.is_integer for a in self.args), quick_exit=True)
+    def _eval_is_integer(self):
+        if len(self.args) == 2:
+            t = a, b = [i.is_integer for i in self.args]
+            if None not in t:
+                if a == b:
+                    if a:
+                        return True
+                elif a != b:
+                    return False
+        else: return _fuzzy_group(
+            (a.is_integer for a in self.args), quick_exit=True)
```

Comment:
Does this change apply to most uses of `_fuzzy_logic`? Maybe we can have a better helper for this.

What exactly is the difference between the change above and not using `quick_exit=True`?

Reply:
I will have to look again tonight or tomorrow. I think group gives up when it finds a T and a F, maybe when quick exit is True? If so, maybe quick exit needs to wait for two F before concluding.

Reply:
False alarm -- not sure how I got onto this.